test/test_sync_subscriptions.cpp

```diff
@@ -844,4 +844,110 @@ TEST(Sync_SyncMetadataSchemaVersions_LegacyTable)
     }
 }
 
+TEST(Sync_MutableSubscriptionSetOperations)
+{
+    SHARED_GROUP_TEST_PATH(sub_store_path);
+    SubscriptionStoreFixture fixture(sub_store_path);
+    auto store = SubscriptionStore::create(fixture.db, [](int64_t) {});
+
+    auto read_tr = fixture.db->start_read();
+    Query query_a(read_tr->get_table("class_a"));
+    query_a.greater_equal(fixture.bar_col, int64_t(1));
+    Query query_b(read_tr->get_table(fixture.a_table_key));
+    query_b.equal(fixture.foo_col, "Realm");
+    Query query_c(read_tr->get_table(fixture.a_table_key));
+
+    // insert_or_assign
+    {
+        auto out = store->get_latest().make_mutable_copy();
+        auto [it, inserted] = out.insert_or_assign("a sub", query_a);
+        CHECK(inserted);
+        auto named_id = it->id;
+        out.insert_or_assign("b sub", query_b);
+        CHECK_EQUAL(out.size(), 2);
+        std::tie(it, inserted) = out.insert_or_assign("a sub", query_a);
+        CHECK(!inserted);
+        CHECK_EQUAL(it->id, named_id);
+        CHECK_EQUAL(out.size(), 2);
+    }
+
+    // find
+    {
+        auto out = store->get_latest().make_mutable_copy();
+        auto [it, inserted] = out.insert_or_assign("a sub", query_a);
+        std::tie(it, inserted) = out.insert_or_assign("b sub", query_b);
+        CHECK(out.find(query_b));
+        CHECK(out.find("a sub"));
+    }
+
+    // erase
+    {
+        auto out = store->get_latest().make_mutable_copy();
+        out.insert_or_assign("a sub", query_a);
+        out.insert_or_assign("b sub", query_b);
+        out.insert_or_assign("c sub", query_c);
+        CHECK_EQUAL(out.size(), 3);
+        auto it = out.erase(out.begin());
+        // Iterator points to last query inserted due do "swap and pop" idiom.
+        CHECK_EQUAL(it->query_string, query_c.get_description());
+        CHECK_EQUAL(out.size(), 2);
+        CHECK(!out.erase("a sub"));
+        CHECK_EQUAL(out.size(), 2);
+        CHECK(out.erase(query_b));
+        CHECK_EQUAL(out.size(), 1);
+        CHECK(out.erase("c sub"));
+        CHECK_EQUAL(out.size(), 0);
+    }
+
+    // erase_by_class_name
```

Comment:
For this and `erase_by_id` add a negative case, where the item isn't found in the list.